spec/system/product_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+describe 'Product' do
+  it 'view products' do
+    # customer = Fabricate(:customer)
+
+    product = Fabricate(:product)
+    login_as product.customer, scope: :customer
+
+    # visit "/product/#{product.id}"
+    # visit product
+    visit product_path(product)
+    expect(page).to have_text(product.name)
+    expect(page).to have_text(product.price)
+    # expect(page).to have_text(product.token)
+  end
+
+  it 'view product details' do
+    product = Fabricate(:product)
+    login_as product.customer, scope: :customer
+
+    visit product_path(product)
+    expect(page).to have_text(product.name)
+    expect(page).to have_text(product.price)
+    expect(page).to have_text(product.token)
+  end
+
+  it 'product is valid with name, price and token' do
+    product = Fabricate.build(:product)
+
+    expect(product).to be_valid
+  end
+
+  it 'product is invalid without attributes' do
+    product = Fabricate.build(:product, name: nil, price: nil)
+    product.valid?
+    expect(product.errors[:name]).to include('não pode ficar em branco')
+    expect(product.errors[:price]).to include('não pode ficar em branco')
+  end
+
+  it 'product is invalid with token duplicate' do
+    Fabricate(:product, token: '12345678901234567890')
+    product = Fabricate.build(:product, token: '12345678901234567890')
+    product.valid?
+    expect(product.errors[:token]).to include('deve ser único')
+  end
+
+  it 'create product' do
+    customer = Fabricate(:customer)
+    visit products_path
+
+    fill_in 'Email', with: customer.email
+    fill_in 'Password', with: customer.password
+
+    click_on 'Log in'
+
+    click_on 'Cadastrar novo Produto'
+
+    fill_in 'Nome', with: 'produto de ponta'
+    fill_in 'Preço', with: '200'
+
+    click_on 'Criar Product'
+
+    assert_current_path product_path(Product.last)
+    expect(page).to have_text('produto de ponta')
+    expect(page).to have_text(200.0)
+  end
+
+  it 'edit product' do
+    customer = Customer.create!(email: 'jane.doe@iugu.com.br', password: 'password')
+    login_as customer, scope: :customer
+    product = Fabricate(:product)
+
+    visit product_path(product)
+
+    new_name = 'produto fera'
+    visit product_path(product)
+
+    click_on 'Editar Product'
+    fill_in 'Nome', with: new_name
+    # assert_current_path
+    click_on 'Atualizar Product'
+    expect(page).to have_content(new_name)
+
+    # assert_current_path
+    expect(page).to have_text('Produto atualizado com sucesso!')
+  end
+
+  it 'delete product' do
+    customer = Customer.create!(email: 'jane.doe@iugu.com.br', password: 'password')
+    login_as customer, scope: :customer
+    product = Fabricate(:product)
+
+    visit product_path(product)
+
+    click_on 'Deletar Product'
+    expect(Product.last).to_not eq product
```

Comment:
Aqui vocês podem usar algo mais claro talvez verificar um nil ou coisa assim.

```suggestion
    expect(Product.last).to be_blank
```